src/Signature/SignatureV4.php

```diff
@@ -82,6 +82,7 @@ public function presign(
         $expires,
         array $options = []
     ) {
+        $expiresTimestamp = $this->convertToTimestamp($expires);
```

Comment:
Isn't this still susceptible to being incorrect? The only way to guarantee the `strtotime` case for `convertExpires()` will be in coordination with the `$startTimestamp` [is to pass in a `$now`](http://php.net/manual/en/function.strtotime.php).

Consider the case:
1:23:45.9999 - `$expiresTimestamp` generated from '+1 week'
1:23:46.0005 - `$startTimestamp` generated via time()
1:23:46:0100 - `convertExpires()` is called, generating 604799